### PR TITLE
Add electron build workflow

### PR DESCRIPTION
**Pull request by azmy60**
In attempt to support electron build as mentioned in #70.

This PR aims to build binaries for electron `18.3.x` to latest.

### CODE REVIEW

File: .github/workflows/Electron.yml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        shell: bash
+        run: ./scripts/node_version.sh
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NODE_AUTH_TOKEN}}
+  
+      - name: Electron ${{ matrix.version.electron }}
+        shell: bash
+        run: ./scripts/node_build.sh 18
+        env: 
+          ELECTRON_VERSION: {{{ matrix.version.electron }}}
```
Comment (carlopi) [inline .github/workflows/Electron.yml:159]:
```suggestion
          ELECTRON_VERSION: ${{ matrix.version.electron }}
```
Probably like this

File: .github/workflows/Electron.yml [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        with:
+          key: ${{ github.job }}-${{ matrix.version.node }}
+          save: ${{ github.ref == 'refs/heads/main' || github.repository != 'duckdb/duckdb-node' }}
+          variant: sccache
+  
+      - name: Electron
+        shell: bash
+        run: ./scripts/node_build_win.sh
+        env:
+          ELECTRON_VERSION: {{{ matrix.version.electron }}}
```
Comment (carlopi) [inline .github/workflows/Electron.yml:264]:
```suggestion
          ELECTRON_VERSION: ${{ matrix.version.electron }}
```

File: .github/workflows/Electron.yml [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+      - name: Validate Docs
+        run: npx jsdoc-to-markdown --files lib/*.js >> $GITHUB_STEP_SUMMARY
+        env:
+          npm_config_yes: true
+
+      - name: Electron ${{ matrix.version.electron }}
+        shell: bash
+        run: ./scripts/node_build.sh ${{ matrix.version.node }}
+        env:
+          ELECTRON_VERSION: {{{ matrix.version.electron }}}
```
Comment (carlopi) [inline .github/workflows/Electron.yml:113]:
```suggestion
          ELECTRON_VERSION: ${{ matrix.version.electron }}
```

File: .github/workflows/Electron.yml [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        shell: bash
+        run: ./scripts/node_version.sh
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NODE_AUTH_TOKEN}}
+  
+      - name: Electron ${{ matrix.version.electron }}
+        shell: bash
+        run: ./scripts/node_build.sh 18
+        env:
+          ELECTRON_VERSION: {{{ matrix.version.electron }}}
```
Comment (carlopi) [inline .github/workflows/Electron.yml:205]:
```suggestion
          ELECTRON_VERSION: ${{ matrix.version.electron }}
```

**Review by carlopi — CHANGES_REQUESTED** [review]
I think those syntax need changing

File: .github/workflows/Electron.yml [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NODE_AUTH_TOKEN}}
+  
+      - name: Electron ${{ matrix.version.electron }}
+        shell: bash
+        run: ./scripts/node_build.sh 18
+        env:
+          ELECTRON_VERSION: ${{ matrix.version.electron }}
+  
+  win-nodejs:
```
Comment (carlopi) [inline .github/workflows/Electron.yml:207]:
```suggestion
  win-electron:
```

**Comment by carlopi** [comment]
Thanks for the contribution, could you have a look at why CI is failing?

**Comment by azmy60** [comment]
@carlopi So it was failing because I needed to add the patch version to the `electron-versions.json`. It should be working now.

**Review by carlopi — CHANGES_REQUESTED** [review]
Thanks, looks good, last request, could you make it so only a small subset (say up to 3?) of the electron builds are actually run on pull requests?

I could also take a look at tweaking this at some later point, but I think having like the setup step be conditional, and either use the full matrix or only selected rows it would be handy to have, since otherwise this will become very heavy.

**Review by carlopi — APPROVED** [review]
Thanks!!

**Comment by azmy60** [comment]
No problem! I just filtered 2 versions that we're currently using in our repo now. :)

**Comment by carlopi** [comment]
@azmy60: given you have it still fresh, could you take a look at the syntax that triggers this problem https://github.com/duckdb/duckdb-node/actions/runs/9937704658: 
 ```
--
Error when evaluating 'strategy' for job 'osx-electron-x64'. .github/workflows/Electron.yml (Line: 167, Col: 18): Error parsing fromJson,.github/workflows/Electron.yml (Line: 167, Col: 18): Error reading JToken from JsonReader. Path '', line 0, position 0.,.github/workflows/Electron.yml (Line: 167, Col: 18): Unexpected value ''Show less
 

<br class="Apple-interchange-newline">
```

**Pull request merged**